### ISSUE TITLE
Incorrect processing of two digit version number in Project.toml

### ISSUE THREAD

**Issue by laughingrice** [JuliaLang/Pkg.jl#3314]
julia> versioninfo()
Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin21.4.0)
  CPU: 4 × Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 4 virtual cores

Installed via "juliaup add release" under Intel mac with juliaup installed via anaconda

Inside Package.toml for OpticSim.jl, under [compat] section, there is an entry:

Pluto = "0.12, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19"

Running:
(@v1.8) pkg> st --outdated -m 

shows:
⌅ [c3e4b0f8] Pluto v0.19.9 (<v0.19.19): OpticSim

I tried with several other packages and variations, but it looks like there is some miss processing based on the last digit instead of the full number after the dot

**Comment by laughingrice** [comment]
The problem turned out to be an issue with verbosity.
OpticSim was limiting the version of JSServe (to 1.*) which seems to be in conflict with the requirement of one of the other packages.
Updating the compatibility to JSServe v2 resolved conflict, but "st --outdated -m" only reported OpticSim as the culprit and not the indirect dependency graph

**Comment by KristofferC** [comment]
Pwew, this sounded scary :).

**Comment by laughingrice** [comment]
Not sure if scary ;-) but extremely difficult to track down.“st —outdated -m” only shows the first branch on the dependency tree, and it is hard to follow the output of force update (up ***@***.***). Some improved clarity on that dependency graph error could help, but that would be a feature request I guess, not a bug. On Jan 15, 2023, at 14:22, Kristoffer Carlsson ***@***.***> wrote:﻿
Pwew, this sounded scary :).
